### ISSUE TITLE
[vNext] Diff colouring not enough contrast

### ISSUE THREAD

**Issue by RussKie** [gitextensions/gitextensions#11821]
### Environment

- Git Extensions 4.3.0.17872-mbjjveco
- Build 70184a4c7252b263427ed627f5004e288e51b4bb (Dirty)
- Git 2.45.2.windows.1

### Issue description

The colours need to have greater contrast ratio (IIRC, it has to be 4:1 or close to it), right now they are too faint
This is an accessibility issue.

![image](https://github.com/user-attachments/assets/f1801fa7-f484-4bea-9811-b14f4e9b8c4c)

### Steps to reproduce

1. Install **GitExtensions-x64-4.3.0.17872-70184a4c7.msi** from https://ci.appveyor.com/project/gitextensions/gitextensions/builds/50328130/artifacts
1. Click on a diff 1cd999e35bd4c7c0eb4cb28671927ddfcf7facad for tests/app/UnitTests/GitCommands.Tests/UserRepositoryHistory/RecentRepoSplitterTests.cs

**Comment by gerhardol** [comment]
This is the default git colors, set by name in git. It is possible to override colors for users with git config (name or color code). Red/green is slightly changed by default in ge (unless user override). 
This is described in the doc pr.

Translation from terminal color names to color code is done as a theme, hardcoded right now. But change one color may require changing 23 other.
As a basis the light theme in putty was used. There were quite few good alternatives. I do not feel I can improve the theme. There are comments in the code about origins.

**Comment by RussKie** [comment]
We're passing the following arguments to `git diff` command setting up colours for the "moved" code:
```
-c color.ui=never -c diff.submodule=short -c diff.noprefix=false -c diff.mnemonicprefix=false -c diff.ignoreSubmodules=none -c core.safecrlf=false -c diff.colorMovedWS=no -c diff.wordRegex=. -c diff.colorMoved=dimmed-zebra -c color.diff.oldMoved=magenta -c color.diff.newMoved=yellow -c color.diff.oldMovedAlternative=cyan -c color.diff.newMovedAlternative=yellow -c color.diff.oldMovedDimmed="magenta dim" -c color.diff.newMovedDimmed="yellow dim" -c color.diff.oldMovedAlternativeDimmed="cyan dim" -c color.diff.newMovedAlternativeDimmed="yellow dim" -c color.diff.old="#000000 #ffc8c8" -c color.diff.new="#000000 #c8ffc8"
```

In https://github.com/gitextensions/gitextensions/pull/11590 you added "ANSI Color theme" section, which describes the colours. 

Is there a reason we only use foreground colour and don't use background for the "moved" sections? A light blue on white has low contrast ratio, whereas a black/white on light blue may yield a much better contrast ratio and be easier to read.
![image](https://github.com/user-attachments/assets/83646d5e-1854-427f-922e-6021e658a70c)

**Comment by gerhardol** [comment]
> Is there a reason we only use foreground colour and don't use background for the "moved" sections? A light blue on white has low contrast ratio, whereas a black/white on light blue may yield a much better contrast ratio and be easier to read.

Because we have not defined theme colors and implemented the overrides for the specific Git color "slots" for moved code.
Personally, I prefer the non inverted colors for the less important moved code.
Note that the GE override colors are not aligned with the terminal theme colors. Chanimg the GE theme color would be a very visible change and change the terminal colors will require a careful alignment of all colors.

I agree that the colors are not optimal. This was discussed in the PR, but did not find tweaking improved the theme in general.
It is possible to disable the coloring engine from GE settings, the algorithm  by setting in Git as well as setting Git color slots, so it can be changed.
See the doc PR for more details: https://github.com/gitextensions/GitExtensionsDoc/pull/151

Maybe the terminal colors should be themeable.
